### PR TITLE
CI/CD Upgrade.

### PR DESCRIPTION
**Pull request by pwei1018**
*Issue #:* https://github.com/bcgov/entity/issues/7578###

*Description of changes:*
1. Deprecate Jenkins pipeline;
2. Handle test and prod CD flow by Makefile and GHA;
3. Enable Github environments for each repo;
4. Using multiple steps check in GHA CI flow;
5. Switch "development" branch to "main" branch;

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-pay license (Apache 2.0).

### CODE REVIEW

File: .github/workflows/pay-api-ci.yml [hunk 1]
```diff
@@ -1,47 +1,69 @@
...
+name: Report API CI
 
 on:
   pull_request:
     branches:
-      - development
+      - main
     paths:
-      - "pay-api/**"
+      - "report-api/**"
```
Comment (sumesh-aot) [inline .github/workflows/pay-api-ci.yml:8]: Isn't this CI for pay-api? all references are for report

File: .github/workflows/payment-reconciliations-cd.yml [hunk 2]
```diff
@@ -3,52 +3,90 @@ name: Payment Reconciliations Queue CD
 on:
   push:
     branches:
-      - development
-      - master
+      - main
     paths:
-      - "queue_services/payment-reconciliations/**"
+      - "queue_services/payment-reconciliation/**"
```
Comment (sumesh-aot) [inline .github/workflows/payment-reconciliations-cd.yml:8]: missing **s**, it's payment-reconciliations

File: .github/workflows/payment-reconciliations-cd.yml [hunk 3]
```diff
@@ -3,52 +3,90 @@ name: Payment Reconciliations Queue CD
...
-        description: "Environment (dev/test)"
+      environment:
+        description: "Environment (dev/test/prod)"
         required: true
         default: "dev"
 
+defaults:
+  run:
+    shell: bash
+    working-directory: ./queue_services/payment-reconciliation
```
Comment (sumesh-aot) [inline .github/workflows/payment-reconciliations-cd.yml:19]: same as above

File: .github/workflows/report-api-ci.yml [hunk 4]
```diff
@@ -1,29 +1,30 @@
-name: Report API CI
+name: PAY API CI
```
Comment (sumesh-aot) [inline .github/workflows/report-api-ci.yml:1]: report and pay CI got reversed I guess

File: .github/workflows/payment-reconciliations-cd.yml (hunk 2 above)
Comment (pwei1018) [reply .github/workflows/payment-reconciliations-cd.yml]: Fixed

File: .github/workflows/pay-api-ci.yml (hunk 1 above)
Comment (pwei1018) [reply .github/workflows/pay-api-ci.yml]: Messed up by copy & paste. Fixed.

File: .github/workflows/report-api-ci.yml (hunk 4 above)
Comment (pwei1018) [reply .github/workflows/report-api-ci.yml]: Yes, Fixed

**Review by sumesh-aot — APPROVED** [review]

**Pull request merged**